dll/win32/msafd/misc/sndrcv.c

```diff
@@ -291,6 +291,13 @@ WSPRecv(SOCKET Handle,
                                    NULL,
                                    0);
 
+    /* Non-blocking sockets must wait until data is available */
+    if (Status == STATUS_PENDING && Socket->SharedData->NonBlocking)
+    {
+        if (lpErrno) *lpErrno = WSAEWOULDBLOCK;
+        return SOCKET_ERROR;
```

Comment:
I wonder whether it could be more consistent to use this:
```suggestion
        return MsafdReturnWithErrno(STATUS_CANT_WAIT, lpErrno, IOSB->Information, lpNumberOfBytesRead);
```
(Sometimes the last two parameters are just set to `0, NULL` ...)

So I know there are at least two other places that just do this direct `lpErrno` manipulation + `return SOCKET_ERROR` in the code, and I was wondering whether using the `MsafdReturnWithErrno` function as done (almost?) everywhere else wouldn't be cleaner.

Can someone investigate this alternative?

Reply:
ok so let's remember that for next time :)